javascript/252-Meeting-Rooms.js

```diff
@@ -1,14 +1,24 @@
 /**
- * @param {number[]} intervals
+ * @param {number[][]} intervals
+ * Time O(N * logN) | Space O(1)
  * @return {boolean}
  */
-var canAttendMeetings = function (intervals) {
-  intervals.sort((a, b) => a[0] - b[0]);
+ var canAttendMeetings = function(intervals) {
+    intervals.sort(([ aStart, aEnd ], [ bStart, bEnd ]) => aStart !== bStart
+        ? aStart - bStart
+        : aEnd - bEnd
+    );
 
-  for (let i = 0; i < intervals.length - 1; i++) {
-    if (intervals[i][1] > intervals[i + 1][0]) {
-      return false;
+    let prev = intervals.shift();
+
+    for (const curr of intervals) {
+        const hasOverlap = ([ prevStart, prevEnd ], [ currStart, currEnd ]) =>
```

Comment:
redefining a function every loop isn't super efficient. maybe try destructuring the `curr` within the loop?

```
for (const [currStart, currEnd] of intervals) {
  const hasOverlap = currStart < prevEnd;
  if (hasOverlap) return false;
  ...
```

Reply:
I will revise this suggestion! It was written to provide context readability over performance